src/OrchardCore/OrchardCore.Data.YesSql/Removing/ShellDbTablesRemovingHandler.cs

```diff
@@ -0,0 +1,171 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Data.Sqlite;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Data;
+using OrchardCore.Data.Documents;
+using OrchardCore.Data.Migration;
+using OrchardCore.Environment.Shell.Builders;
+using OrchardCore.Environment.Shell.Models;
+using OrchardCore.Environment.Shell.Scope;
+using YesSql;
+
+namespace OrchardCore.Environment.Shell.Removing;
+
+/// <summary>
+/// Allows to remove the database tables retrieved from the migrations of a given tenant.
+/// </summary>
+public class ShellDbTablesRemovingHandler : IShellRemovingHandler
+{
+    private readonly IShellContextFactory _shellContextFactory;
+    private readonly IStringLocalizer S;
+    private readonly ILogger _logger;
+
+    public ShellDbTablesRemovingHandler(
+        IShellContextFactory shellContextFactory,
+        IStringLocalizer<ShellDbTablesRemovingHandler> localizer,
+        ILogger<ShellDbTablesRemovingHandler> logger)
+    {
+        _shellContextFactory = shellContextFactory;
+        S = localizer;
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// Removes the database tables retrieved from the migrations of the provided tenant.
+    /// </summary>
+    public async Task RemovingAsync(ShellRemovingContext context)
+    {
+        if (context.LocalResourcesOnly)
+        {
+            return;
+        }
+
+        if (context.ShellSettings.State == TenantState.Uninitialized)
+        {
+            // An 'Uninitialized' tenant with a valid connection, and at least the 'Document' table,
```

Comment:
I think these 2lines of comments can be removed. later on in the code you are documenting what happens

Reply:
Yes, it is an additional comment for the special case of an `Uninitialized` whose setup failed.

Okay will see but the following comments don't talk about a failing setup, explaining that some tables may need to be cleaned up, even if the tenant is `Uninitialized`.

Reply:
Okay, removed these comments locally.